test/irmin-pack/concurrent.ml

```diff
@@ -0,0 +1,191 @@
+    () )
+
+(* Helpers for Unix pipe *)
+let write input =
+  try
+    let m = Bytes.of_string (string_of_int (Unix.getpid ())) in
+    let n = Unix.write input m 0 (Bytes.length m) in
+    assert (n = Bytes.length m)
+  with Unix.Unix_error (n, f, arg) ->
+    Alcotest.failf "Unix error %s %s %s " f arg (Unix.error_message n)
+
+let read output =
+  let buff = Bytes.create 5 in
+  match Unix.read output buff 0 5 with
+  | 0 -> Alcotest.fail "Something wrong when reading from the pipe"
+  | n -> int_of_string (Bytes.to_string (Bytes.sub buff 0 n))
+
+let write_hash input h =
+  let hash_size = Hash.hash_size in
+  try
+    let buf = Bytes.of_string (Irmin.Type.to_bin_string Hash.t h) in
+    assert (hash_size = Bytes.length buf);
+    let n = Unix.write input buf 0 (Bytes.length buf) in
+    assert (n = Bytes.length buf)
+  with Unix.Unix_error (n, f, arg) ->
+    Alcotest.failf "Unix error %s %s %s " f arg (Unix.error_message n)
+
+let read_hash output =
+  let hash_size = Hash.hash_size in
+  let buf = Bytes.create hash_size in
+  let n = Unix.read output buf 0 (Bytes.length buf) in
+  assert (n = Bytes.length buf);
+  match Irmin.Type.of_bin_string Hash.t (Bytes.to_string buf) with
+  | Ok t -> t
+  | Error (`Msg e) -> Fmt.failwith "read hash: %s" e
+
+let wait pid =
+  Lwt_unix.waitpid [ Unix.WUNTRACED ] pid >>= fun (pid', status) ->
+  if pid <> pid' then
+    Alcotest.failf "I'm %d, expecting child %d, but got %d instead"
+      (Unix.getpid ()) pid pid';
+  match status with
+  | Unix.WEXITED 0 ->
+      Log.debug (fun l -> l "Child %d finished work" pid);
+      Lwt.return_unit
+  | Unix.WSTOPPED s ->
+      Alcotest.failf "Child %d died unexpectedly stopped by %d" pid s
+  | Unix.WEXITED s ->
+      Alcotest.failf "Child %d died unexpectedly exited with %d" pid s
+  | Unix.WSIGNALED s ->
+      Alcotest.failf "Child %d died unexpectedly signaled by %d" pid s
+
+let nb_batch_writes = 5
+
+let branch i = Printf.sprintf "b%d" i
+
+let key i = Printf.sprintf "a%d" i
+
+let value i = Printf.sprintf "x%d" i
+
+let test_find_present r n h =
+  Log.debug (fun l -> l "[%d] Checking round %d" (Unix.getpid ()) n);
+  let rec aux i =
+    if i = n then Lwt.return_unit
+    else
+      S.Commit.of_hash r h >>= function
+      | None -> Alcotest.fail "no hash"
+      | Some commit ->
+          let msg = Printf.sprintf "RO find %d %d" n i in
+          let tree = S.Commit.tree commit in
+          S.Tree.find tree [ key i ] >>= fun x ->
+          Alcotest.(check (option string)) msg (Some (value i)) x;
+          aux (i + 1)
+  in
+  aux 0
+
+let add rw tree i =
+  Log.debug (fun l -> l "[%d] Adding round %d" parent_pid i);
+  S.Tree.add tree [ key i ] (value i) >>= fun tree ->
+  S.Commit.v rw ~parents:[] ~info:(info ()) tree >|= fun c ->
+  Log.debug (fun l -> l "Commit %a" S.Commit.pp_hash c);
+  (tree, c)
+
+let worker w_pipe r_pipe =
+  Log.debug (fun l -> l "Worker started %d" (Unix.getpid ()));
+  ignore (read r_pipe);
```

Comment:
What's being read and ignored here?

Reply:
[The parent pid](https://github.com/mirage/irmin/pull/1008/files#diff-4fdf416b755e60ec1edde6f08f834590R115). Changed it to be more specific.